rio_stac/stac.py

```diff
@@ -84,6 +86,33 @@ def get_projection_info(
     return meta
 
 
+def get_eobands_info(
+    src_dst: Union[DatasetReader, DatasetWriter, WarpedVRT, MemoryFile]
+) -> List:
+    """Get eo:bands metadata.
+
+    see: https://github.com/stac-extensions/eo#item-properties-or-asset-fields
+
+    """
+    eo_bands = []
+
+    colors = src_dst.colorinterp
+    for ix in src_dst.indexes:
+        band_meta = {"name": f"b{ix}"}
+
+        descr = src_dst.descriptions[ix - 1]
+        color = colors[ix - 1].name
+
+        # Description metadata or Colorinterp or Nothing
+        description = descr or color
+        if description:
+            band_meta["description"] = description
```

Comment:
`eo:bands` description will be either:
- GDAL description metadata 
- color interp value https://github.com/rasterio/rasterio/blob/main/rasterio/enums.py#L26-L45